src/reactiveData.mli

```diff
@@ -17,66 +17,216 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  *)
 
-module type DATA = sig
-  type 'a data
-  type 'a patch
-  val merge : 'a patch -> 'a data -> 'a data
-  val map_patch : ('a -> 'b) -> 'a patch -> 'b patch
-  val map_data : ('a -> 'b) -> 'a data -> 'b data
-  val empty : 'a data
-  val equal : ('a -> 'a -> bool) -> 'a data -> 'a data -> bool
-  val diff : eq:('a -> 'a -> bool) -> 'a data -> 'a data -> 'a patch
-end
+(** ReactiveData is a module for data-aware functional reactive
+    programming (FRP). It adds support to incremental changes in data
+    structures by reasoning on patches instead of absolute
+    values. ReactiveData is based on and inter-operates with React.
+
+    You are most likely interested in the sub-module [RList], which
+    implements a superset of the signature [S]. *)
+
+(** Signature describing a reactive data structure (['a t]).
+
+    Most functions in [S] are not safe to call during a React update
+    step. *)
 module type S = sig
+
+  (** Reactive version of the data container *)
+  type 'a t
+
+  (** Raw (non-reactive) version of the data container *)
   type 'a data
+
+  (** Patch format *)
   type 'a patch
-  type 'a msg = Patch of 'a patch | Set of 'a data
+
+  (** Message format *)
+  type 'a msg =
+    | Patch of 'a patch (** [Patch p] triggers the application of [p]
+                            on the current contents *)
+    | Set of 'a data    (** With [Set d], [d] becomes the new
+                            content *)
+
+  (** Handle that permits applying incremental updates *)
   type 'a handle
-  type 'a t
+
+  (** Empty data structure *)
   val empty : 'a t
+
+  (** Build a container from initial contents. The handle can be used
+      for performing reactive updates. *)
   val make : 'a data -> 'a t * 'a handle
+
+  (** [make_from d e] is a container whose initial value is [d], and
+      which gets updated for every occurrence of [e] *)
   val make_from : 'a data -> 'a msg React.E.t -> 'a t
```

Comment:
`from_event` ? This way  there is a correspondance between event/signal and from_event/from_signal.